tests/app/celery/test_process_ses_receipts_tasks.py

```diff
@@ -282,47 +291,94 @@ def test_ses_callback_should_update_multiple_notification_status_sent(
     assert send_mock.called
 
 
-def test_ses_callback_should_set_status_to_temporary_failure(client, mocker, sample_template, sample_notification):
+@pytest.mark.parametrize('status', [NOTIFICATION_DELIVERED, NOTIFICATION_SENDING, NOTIFICATION_SENT])
+def test_ses_callback_should_set_status_to_temporary_failure(
+    client,
+    mocker,
+    sample_template,
+    sample_notification,
+    sample_service_callback,
+    status,
+):
     send_mock = mocker.patch('app.celery.service_callback_tasks.send_delivery_status_to_service.apply_async')
 
     template = sample_template(template_type=EMAIL_TYPE)
     ref = str(uuid4())
     notification_id = sample_notification(
         template=template,
-        status='sending',
+        status=status,
         reference=ref,
     ).id
 
-    create_service_callback_api(service=template.service, url='https://original_url.com')
+    sample_service_callback(service=template.service, url='https://original_url.com')
 
-    assert process_ses_receipts_tasks.process_ses_results(ses_soft_bounce_callback(reference=ref))
-    assert get_notification_by_id(notification_id).status == 'temporary-failure'
-    assert (
-        get_notification_by_id(notification_id).status_reason
-        == 'Temporarily failed to deliver email due to soft bounce'
-    )
+    assert process_ses_receipts_tasks.process_ses_results(ses_soft_bounce_callback(reference=ref)) is None
+    db_notification = get_notification_by_id(notification_id)
```

Comment:
Given that this is test code, should we not use `notify_db_session.session.get(...)` directly instead of calling dao code?

Reply:
They were there previously, so I continued the trend, but I think it's better to query the DB so they have been changed.